### ISSUE TITLE
Usage Instructions

### ISSUE THREAD

**Issue by dillon-co** [mathieulaporte/machine#1]
This is really cool! 

Can you add some usage instructions for us mere mortals? :)

**Comment by bararchy** [comment]
@grandmasterspoc Guess project is dead ? :\

**Comment by dillon-co** [comment]
@bararchy Yeah looks like it, which is a bummer because I wanted to use it haha

**Comment by bararchy** [comment]
@grandmasterspoc @mathieulaporte .
I'm trying to have some ANN in Crystal working at: https://github.com/bararchy/crystal-ann.

If @mathieulaporte isn't going to maintain this project (which is a shame :\ ) I will take the ANN from https://github.com/mathieulaporte/machine/blob/master/src/machine/neural_network/perceptron.cr or parts of it and will add to my project (both licensed under MIT and I'll give credits of curse) .

I would really love to know though if this is ok from @mathieulaporte point of view.

**Comment by mathieulaporte** [comment]
Hello guys sorry I do not see all notifications in the project (my bad).

I recently have done some work on the project, i will publish it soon (some refactor and new algo).
I only can work on it some weekends...

Machine is just a simple project i work on for personal purpose (learning crystal and machine learning algorithm)

@bararchy This file https://github.com/mathieulaporte/machine/blob/master/src/machine/neural_network/perceptron.cr  is basically a port from this one : 
https://github.com/SergioFierens/ai4r/blob/master/lib/ai4r/neural_network/backpropagation.rb
It is named perceprton.cr but it is a backpropagation neural network. You should give all credit to SergioFierens. This file is going to be removed from the project (even from the history).

I will continue to maintain the project but I will focus on other kind of machine learning algorithm than NN. Bayesian network and decision tree for now ;-)

@grandmasterspoc let me know what you need (Any way I will provide basic examples on Bayesian network this weekend)

**Comment by bararchy** [comment]
@mathieulaporte Thanks for the info.
I took the file to my repo, gave credit both to you and the original repo\owner.

I would really love an Bayes classifier to be available in Crystal, it would be awesome, I even wanted to start doing it by porting over [classifier-reborn](https://github.com/jekyll/classifier-reborn/issues/100)

- If you need any help I would love to join in on this  
- I would love to get help on the ANN side too, if you want too :)

**Comment by dillon-co** [comment]
@mathieulaporte Thanks! I was actually hoping for some instructions on the ANN. But I'm trying to learn how to write my own ML algorithms so I'll definitely be taking a look at the source :)
